checker.c

```diff
@@ -1,21 +1,52 @@
-#include <stdio.h>
-#include <assert.h>
+#include "checker.h"
+
+float readchargeRate(float chargeRate)
+{
+    if(chargeRate > CHARGE_RATE_MAX)
+    {
+        printf("Charge Rate out of range!\n");
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+float readsoc(float soc)
+{
+    if(soc < SOC_MIN || soc > SOC_MAX)
```

Comment:
similar code check in temperature too. scope for improvement by reducing code duplication...